From 612c45821f32b2e6602aec64413b3513fc2734df Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:34 -0400
Subject: [tools] lib/traceevent: Get rid of malloc_or_die() in show_error()

Message-id: <1408953210-26343-147-git-send-email-jolsa@redhat.com>
Patchwork-id: 88842
O-Subject: [PATCH RHEL7.1 BZ1133083 146/322] tools lib traceevent: Get rid of malloc_or_die() in show_error()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 9451a2fd78c785445afe0f6966b2043c3ee187ca
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Dec 12 16:36:04 2013 +0900
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/parse-filter.c b/tools/lib/traceevent/parse-filter.c
index ab402fb..d4b0bac 100644
--- a/tools/lib/traceevent/parse-filter.c
+++ b/tools/lib/traceevent/parse-filter.c
@@ -56,7 +56,21 @@ static void show_error(char **error_str, const char *fmt, ...)
  index = pevent_get_input_buf_ptr();
  len = input ? strlen(input) : 0;
 
- error = malloc_or_die(MAX_ERR_STR_SIZE + (len*2) + 3);
+ error = malloc(MAX_ERR_STR_SIZE + (len*2) + 3);
+ if (error == NULL) {
+  /*
+   * Maybe it's due to len is too long.
+   * Retry without the input buffer part.
+   */
+  len = 0;
+
+  error = malloc(MAX_ERR_STR_SIZE);
+  if (error == NULL) {
+   /* no memory */
+   *error_str = NULL;
+   return;
+  }
+ }
 
  if (len) {
   strcpy(error, input);
-- 
1.7.1